packages/cozy-harvest-lib/src/datacards/ViewerModal.jsx

```diff
@@ -0,0 +1,33 @@
+import React, { useContext } from 'react'
+import Modal from '@material-ui/core/Modal'
+
+import Portal from 'cozy-ui/transpiled/react/Portal'
+import Viewer from 'cozy-ui/transpiled/react/Viewer'
+
+import { MountPointContext } from '../components/MountPointContext'
+import { useDataCardFiles } from './makeViewerFiles'
+
+export const ViewerModal = ({
+  match: {
+    params: { accountId, triggerId, fileIndex }
+  }
+}) => {
+  const { pushHistory, replaceHistory } = useContext(MountPointContext)
+  const handleCloseViewer = () => replaceHistory(`/accounts`)
+  const handleFileChange = (_file, newIndex) =>
+    pushHistory(`/viewer/${accountId}/${triggerId}/${newIndex}`)
+  const files = useDataCardFiles(accountId, triggerId)
+
+  return files && files.length > 0 ? (
+    <Portal into="body">
+      <Modal open={true}>
+        <Viewer
+          files={files}
+          currentIndex={Number(fileIndex)}
+          onCloseRequest={handleCloseViewer}
+          onChangeRequest={handleFileChange}
+        />
+      </Modal>
+    </Portal>
+  ) : null
```

Comment:
Shouldn't we display an error message or something if there is no files to display ? What is the current behavior with this implementation?

Reply:
Well, there's none, to be honest, but here it's more a case of I don't know what should I do in terms of behavior. What do we want?